l1-contracts/contracts/state-transition/StateTransitionManager.sol

```diff
@@ -124,28 +124,55 @@ contract StateTransitionManager is IStateTransitionManager, ReentrancyGuard, Own
         require(_initializeData.owner != address(0), "STM: owner zero");
         _transferOwnership(_initializeData.owner);
 
-        genesisUpgrade = _initializeData.genesisUpgrade;
         protocolVersion = _initializeData.protocolVersion;
         protocolVersionDeadline[_initializeData.protocolVersion] = type(uint256).max;
         validatorTimelock = _initializeData.validatorTimelock;
 
+        _setChainCreationParams(_initializeData.chainCreationParams);
+
+        // While this does not provide a protection in the production, it is needed for local testing
+        // Length of the L2Log encoding should not be equal to the length of other L2Logs' tree nodes preimages
+        assert(L2_TO_L1_LOG_SERIALIZE_SIZE != 2 * 32);
```

Comment:
I know it is not your change but this assert makes sense to add only in constructor (contract won't deploy rather than some function revert)